app/services/subscriptions/free_trial_billing_service.rb

```diff
@@ -31,7 +31,13 @@ def call
     # during the customer trial period
     # Unfortunately, this introduces an N+1 query
     def already_billed_on_day_one?(subscription)
-      subscription.invoice_subscriptions.where(timestamp: subscription.started_at.all_day).exists?
+      Fee.subscription_kind.where(
+        invoice_id: subscription.invoice_subscriptions.select('invoices.id').joins(:invoice).where(
+          'invoices.invoice_type' => :subscription,
+          'invoices.status' => %i[draft finalized],
+          timestamp: subscription.started_at.all_day,
```

Comment:
Not new, but I'm not sure it exactly right if customer timezone is not UTC :thinking:

Reply:
Very good point, I keep thinking about this and it's still unclear 🙈